cli/root.go

```diff
@@ -430,6 +427,15 @@ type RootCmd struct {
 }
 
 func addTelemetryHeader(client *codersdk.Client, inv *clibase.Invocation) {
+	transport, ok := client.HTTPClient.Transport.(*headerTransport)
+	if !ok {
+		transport = &headerTransport{
```

Comment:
I'm a bit worried about losing headers in a situation where we think it's safe to wrap the transport and we end up with a type that's `headerTransport(otherTransport(headerTransport))`. Not sure what the best approach to guard against this would be, though.